### PR TITLE
Improved automatic certificate generation log and fix bugs

### PR DESCRIPTION
**Pull request by ls889**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

/kind bug
/kind cleanup

**What this PR does / why we need it**:

1.Delete useless functions
2.Add and modify some comments for golint
3.Printed logs more clearly to locate errors
4.Now there are some old codes of RSA, I have fixed them all in the pr.
5.Automatically create kubeedge namespace when creating a secret

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cloud/pkg/cloudhub/servers/httpserver/certutil.go [hunk 1]
```diff
@@ -59,6 +58,7 @@ func NewSelfSignedCACertDERBytes(key crypto.Signer) ([]byte, error) {
 	return caDERBytes, err
 }
 
+// NewCloudCoreCertDERandKey creates CloudCore certificate and key
```
Comment (fisherxu) [inline cloud/pkg/cloudhub/servers/httpserver/certutil.go:61]: NewCloudCoreCertDERandKey creates certificate and key for CloudCore

File: cloud/pkg/cloudhub/servers/httpserver/secretsutil.go [hunk 2]
```diff
@@ -3,7 +3,11 @@ package httpserver
 import (
 	"fmt"
 
+	"k8s.io/client-go/kubernetes"
+
```
Comment (fisherxu) [inline cloud/pkg/cloudhub/servers/httpserver/secretsutil.go:7]: delete the empty line

File: cloud/pkg/cloudhub/servers/httpserver/secretsutil.go [hunk 3]
```diff
@@ -27,7 +31,7 @@ const (
 func GetSecret(secretName string, ns string) (*v1.Secret, error) {
 	cli, err := utils.KubeClient()
 	if err != nil {
-		fmt.Printf("%v", err)
+		klog.Warningf("Failed to create KubeClient, error: %s", err)
```
Comment (fisherxu) [inline cloud/pkg/cloudhub/servers/httpserver/secretsutil.go:34]: Do we need return here?

File: cloud/pkg/cloudhub/servers/httpserver/secretsutil.go [hunk 4]
```diff
@@ -36,7 +40,10 @@ func GetSecret(secretName string, ns string) (*v1.Secret, error) {
 func CreateSecret(secret *v1.Secret, ns string) error {
 	cli, err := utils.KubeClient()
 	if err != nil {
-		fmt.Printf("%v", err)
+		klog.Warningf("Failed to create KubeClient, error: %s", err)
```
Comment (fisherxu) [inline cloud/pkg/cloudhub/servers/httpserver/secretsutil.go:43]: ditto

File: cloud/pkg/cloudhub/servers/httpserver/secretsutil.go [hunk 5]
```diff
@@ -36,7 +40,10 @@ func GetSecret(secretName string, ns string) (*v1.Secret, error) {
...
+	}
+	if err := CreateNamespaceIfNeeded(cli, ns); err != nil {
+		klog.Warningf("Failed to create Namespace kubeedge, error: %s", err)
```
Comment (fisherxu) [inline cloud/pkg/cloudhub/servers/httpserver/secretsutil.go:46]: ditto

File: cloud/pkg/cloudhub/servers/httpserver/secretsutil.go [hunk 6]
```diff
@@ -98,3 +105,21 @@ func CreateCloudCoreSecret(certDER, key []byte) error {
...
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ns,
+			Namespace: "",
+		},
+	}
+	_, err := c.Namespaces().Create(newNs)
+	if err != nil && errors.IsAlreadyExists(err) {
+		err = nil
+	}
+	return err
```
Comment (fisherxu) [inline cloud/pkg/cloudhub/servers/httpserver/secretsutil.go:124]: Please refer to: https://github.com/kubeedge/kubeedge/blob/master/keadm/cmd/keadm/app/cmd/util/k8sinstaller.go#L73

File: cloud/pkg/cloudhub/servers/httpserver/signcerts.go [hunk 7]
```diff
@@ -70,25 +71,32 @@ func GenerateToken() {
...
 	}
 
 	caHash := getCaHash()
 	// combine caHash and tokenString into caHashAndToken
 	caHashToken := strings.Join([]string{caHash, tokenString}, ".")
 	// save caHashAndToken to secret
-	CreateTokenSecret([]byte(caHashToken))
+	err = CreateTokenSecret([]byte(caHashToken))
+	if err != nil {
+		klog.Errorf("Failed to create tokenSecret, err: %v", err)
```
Comment (fisherxu) [inline cloud/pkg/cloudhub/servers/httpserver/signcerts.go:83]: Do we need return the err？

**Comment by fisherxu** [comment]
And please rebase the master branch to fix the CI issue.

File: cloud/pkg/cloudhub/servers/httpserver/secretsutil.go (hunk 6 above)
Comment (ls889) [reply cloud/pkg/cloudhub/servers/httpserver/secretsutil.go:124]: I looked at these two functions and they are the same, the function CreateNamespaceIfNeeded is from k8s.

Comment (fisherxu) [reply cloud/pkg/cloudhub/servers/httpserver/secretsutil.go:122]: just keep it here.

**Comment by ls889** [comment]
Kindly ping @fisherxu 
thanks for suggestions, Done.

**Pull request merged**